js/logic/big-picture/big-picture-comment.js

```diff
@@ -1,19 +1,81 @@
+const bigPicture = document.querySelector('.big-picture');
+const commentsContainer = bigPicture.querySelector('.social__comments');
+const commentsInfoContainer = bigPicture.querySelector('.social__comment-count');
+const commentsCurrentCountContainer = commentsInfoContainer.querySelector('.social__comment-current-count');
+const commentsAllCountContainer = commentsInfoContainer.querySelector('.comments-count');
+const loadingButton = bigPicture.querySelector('.social__comments-loader');
+
 const userCommentTemplate = document.querySelector('#user-comment')
   .content
   .querySelector('.social__comment');
 
-export const createUserCommentElement = ({
+const createUserComment = ({
   avatar: avatarSrc,
   name: username,
   message: userText,
 }) => {
-  const userCommentElement = userCommentTemplate.cloneNode(true);
+  const userComment = userCommentTemplate.cloneNode(true);
 
-  const avatar = userCommentElement.querySelector('.social__picture');
+  const avatar = userComment.querySelector('.social__picture');
   avatar.src = avatarSrc;
   avatar.alt = username;
 
-  userCommentElement.querySelector('.social__text').textContent = userText;
+  userComment.querySelector('.social__text').textContent = userText;
+
+  return userComment;
+};
+
+let loadingButtonClickHandler = null;
+export const removeLoadingButtonClickHandler = () => {
+  loadingButton.removeEventListener('click', loadingButtonClickHandler);
+};
+
+const addNewComments = (comments, startIndex, commentsCount) => {
+  const lastIndex = startIndex + commentsCount;
+
+  commentsContainer.append(
+    ...comments
+      .slice(startIndex, lastIndex)
+      .map(createUserComment)
+  );
+
+  if (lastIndex >= comments.length) {
+    loadingButton.classList.add('hidden');
+    removeLoadingButtonClickHandler();
+  }
+};
+
+const addLoadingButtonClickHandler = (comments, startIndex = 0, commentsCount) => {
+  if (loadingButtonClickHandler) {
+    removeLoadingButtonClickHandler();
+  }
+
+  loadingButtonClickHandler = () => {
+    addNewComments(comments, startIndex, commentsCount);
+
+    startIndex += commentsCount;
+    startIndex = (startIndex > comments.length) ? comments.length : startIndex;
+
+    commentsCurrentCountContainer.textContent = startIndex;
+  };
+
+  loadingButton.addEventListener('click', loadingButtonClickHandler);
+};
+
+export const fillUserComments = (
+  comments,
+  startCount = 5,
+  newCommentsCount = Infinity,
+) => {
+  const commentsCount = comments.length;
+
+  commentsCurrentCountContainer.textContent = startCount > commentsCount
+    ? commentsCount
+    : startCount;
+  commentsAllCountContainer.textContent = commentsCount;
+  commentsContainer.replaceChildren();
+  loadingButton.classList.remove('hidden');
 
-  return userCommentElement;
+  addNewComments(comments, null, startCount);
```

Comment:
<details>
<summary>logic/big-picture/big-picture-comment.js</summary>

https://github.com/htmlacademy-javascript/2209003-kekstagram-28/blob/d129cf85d35214381df952a5653e6d74a49b1268/js/logic/big-picture/big-picture-comment.js#L33
https://github.com/htmlacademy-javascript/2209003-kekstagram-28/blob/d129cf85d35214381df952a5653e6d74a49b1268/js/logic/big-picture/big-picture-comment.js#L54
https://github.com/htmlacademy-javascript/2209003-kekstagram-28/blob/d129cf85d35214381df952a5653e6d74a49b1268/js/logic/big-picture/big-picture-comment.js#L79
</details>

1. Такое изменение типа параметра приведет к созданию JIT-компилятором двух реализаций этой функции, а представление `startIndex` в памяти будет необратимо ухудшено;

2. Представление о `null` как об отсутствие всякого значение — простая (неточная) ментальная модель, которую транслируют начинающим. But null mean "no object": 


![image](https://user-images.githubusercontent.com/700213/221021994-d2c0c80b-9b71-4d36-b2c3-7bff18fa3675.png)

Актуальная спецификация ecma262:

![image](https://user-images.githubusercontent.com/700213/221022802-deb6facd-6a5b-45d2-8ba3-b4315f565d30.png)

Блог разработчиков движка V8:

![image](https://user-images.githubusercontent.com/700213/221023004-a23870ea-3f6a-4d44-a8ca-b97e6420bc00.png)

> В свете этого знания оказывается, что `null` означает «отсутствие объектного значения», в то время как `undefined` — это «отсутствие значения»

> Следуя этим размышлениям в духе Java, Брендан Эйх спроектировал JavaScript так, чтобы оператор `typeof` возвращал бы `'object'` для значений тех типов, которые расположены на предыдущем рисунке справа. Сюда попадают все объектные значения и `null`. Именно поэтому истинным является выражение `typeof null === 'object'` несмотря на то, что в спецификации языка имеется отдельный тип `Null`

Известный разработчик движка V8, Benedikt Meurer, предупреждает:

> if you intended to operate on number values really, then mixing in non-number values, including `undefined` or `null` primitives **is a bad idea**

> **Try to avoid mixing field values of different types**, i.e. don’t mix numbers, strings, objects, other primitives, unless that’s what you intended to do. Specifically **don’t pre-initialize number fields to `null` or `undefined`**, but choose sensible default numbers (use `NaN` if in doubt)